sklearn/neighbors/tests/test_lmnn.py

```diff
@@ -0,0 +1,377 @@
+import numpy as np
+
+from sklearn.model_selection import train_test_split
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_allclose
+from sklearn.utils.testing import assert_raises
+from sklearn.utils.testing import assert_warns
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_true
+from sklearn.utils.testing import assert_false
+from sklearn import datasets
+from sklearn.neighbors import LargeMarginNearestNeighbor, KNeighborsClassifier
+
+
+rng = np.random.RandomState(0)
+# load and shuffle iris dataset
+iris = datasets.load_iris()
+perm = rng.permutation(iris.target.size)
+iris.data = iris.data[perm]
+iris.target = iris.target[perm]
```

Comment:
I would prefer creating an "iris_data" and "iris_target" variable, to make it more clear to the casual reader who is reading test code, that this is not the mere output of load_iris. (same thing for digits below)

Reply:
Done.